pkg/binding/annotationmapper.go

```diff
@@ -45,26 +47,44 @@ func NewDefinitionBuilder(annotationName string, annotationValue string, configM
 	}
 }
 
-func (m *annotationBackedDefinitionBuilder) outputName() (string, error) {
-	// bail out in the case the annotation name doesn't start with "service.binding"
-	if m.name != AnnotationPrefix && !strings.HasPrefix(m.name, AnnotationPrefix+"/") {
-		return "", fmt.Errorf("can't process annotation with name %q", m.name)
+func (m *annotationBackedDefinitionBuilder) isServiceBindingAnnotation() (bool, error) {
+	if m.name == ProvisionedServiceAnnotationKey {
+		return false, nil
+	}
+
+	if m.name == AnnotationPrefix {
+		return true, nil
+	} else if strings.HasPrefix(m.name, AnnotationPrefix) {
+		if strings.HasPrefix(m.name, AnnotationPrefix+"/") {
+			return true, nil
+		}
+
+		// it starts with AnnotationPrefix, but has extra text at the end not
+		// separated by a /, so treat it as an error
+		return false, fmt.Errorf("can't process annotation with name %q", m.name)
```

Comment:
I would not be so rigid, here is what should return `true`:
* `service.binding`
* `service.binding/xxx`
everything else should return `false`, and I do not think that we case when we should return error.

Reply:
`service.binding` is handled explicitly in [line 55](https://github.com/redhat-developer/service-binding-operator/pull/1023/commits/a27aa032c20919ed62363578c9703c56bca3771d#diff-73e8f8345b43001c4d65bc139e25b3ed9b17a921772b50a3f0a0a125f0758a60R55), and `service.binding/xxx` is handled by [line 58](https://github.com/redhat-developer/service-binding-operator/pull/1023/commits/a27aa032c20919ed62363578c9703c56bca3771d#diff-73e8f8345b43001c4d65bc139e25b3ed9b17a921772b50a3f0a0a125f0758a60R58).  If we've reached here, then we've hit an annotation that looks like `service.bindingxxx`.  Previous behavior here was to throw an error, and I'm preserving that behavior.

Reply:
thanks for the clarification, I missed that.